src/theory/sets/theory_sets_private.cpp

```diff
@@ -1485,10 +1485,55 @@ void TheorySetsPrivate::preRegisterTerm(TNode node)
   }
 }
 
-Node TheorySetsPrivate::expandDefinition(LogicRequest& logicRequest, Node n)
+Node TheorySetsPrivate::expandDefinition(LogicRequest& logicRequest, Node node)
 {
-  Debug("sets-proc") << "expandDefinition : " << n << std::endl;
-  return n;
+  Debug("sets-proc") << "expandDefinition : " << node << std::endl;
+
+  if (node.getKind() == kind::CHOOSE)
+  {
+    // (choose A) is expanded as
+    // (witness ((x elementType)) (and (member x A) (= x chooseUf(A))))
```

Comment:
It should be (and (=> (not (= A emptyset)) (member x A)) (= x chooseUf(A)))

Reply:
done